### PR TITLE
Add note for revision vs commitID [ci skip] 

### PR DESCRIPTION
**Pull request by llewellyn-sl**
Per FD#2725, added a note explaining the difference in behavior between passing a git tag or branch name vs. passing a commit ID using the `-r` argument. 

Signed-off-by: Llewellyn vd Berg <113503285+llewellyn-sl@users.noreply.github.com>

### CODE REVIEW

File: docs/metadata.rst [hunk 1]
```diff
@@ -62,6 +62,8 @@ manifest                    Entries of the workflow manifest.
 | Properties marked with a `*` are accessible only in the workflow completion and error handlers. See the `Completion handler`_ section for details.
 |
 
+.. note:: 
+    When passing a git tag or branch name using the ``-r`` argument in your CLI command, the ``workflow.revision`` and the associated ``workflow.commitId`` is populated. When passing only the Git commit ID using ``-r``, no ``workflow.revision`` is returned.
```
Comment (pditommaso) [inline docs/metadata.rst:66]:
```suggestion
    When passing a Git tag or branch name using the ``-r`` argument in your CLI command, the ``workflow.revision`` and the associated ``workflow.commitId`` is populated. When passing only the Git commit ID using ``-r``, no ``workflow.revision`` is returned. 
```

Comment (pditommaso) [reply docs/metadata.rst:66]: Should not be "are populated" ?

Comment (llewellyn-sl) [inline docs/metadata.rst:66]:
```suggestion
    When passing a Git tag or branch name using the ``-r`` argument in your CLI command, the ``workflow.revision`` and the associated ``workflow.commitId`` are populated. When passing only the Git commit ID using ``-r``, no ``workflow.revision`` is returned. 
```

Comment (llewellyn-sl) [reply docs/metadata.rst]: Certainly. Thanks for catching.

**Review by pditommaso — APPROVED** [review]

**Pull request merged**